src/blocks-transactions-loader/blocks-transactions-loader.ts

```diff
@@ -31,76 +31,66 @@ export class BlocksTransactionsLoader {
     const deleteBlockMinedPayloads: DeleteBlockMinedEventOptions[] = [];
     const upsertBlockMinedPayloads: UpsertBlockMinedEventOptions[] = [];
 
-    const response = await this.prisma.$transaction(
-      async (prisma) => {
```

Comment:
Do you think we should have a transaction per block in this upsert loop to make operations for each block and associated transactions atomic?

Reply:
as discussed, added a comment to address